### ISSUE TITLE
operator_usage_whitespace treats `>` in generic specialization as operator when initializing

### ISSUE THREAD

**Issue by rastersize** [realm/SwiftLint#1897]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

The `operator_usage_whitespace` rule matches the trailing `>` when specializing a generic type and initializing it at the same time.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
  * **0.23.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
  * **Homebrew**
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Which Xcode version are you using (check `xcode-select -p`)?
  * **Xcode 9.0 (final release)**
* Do you have a sample that shows the issue? Yup, run: `echo "internal let foo = GenericType<(UIViewController) -> Void>()" | swiftlint lint --no-cache --use-stdin --enable-all-rules`

```shell
$ echo "internal let foo = GenericType<(UIViewController) -> Void>()" | swiftlint lint --no-cache --use-stdin --enable-all-rules
<nopath>:1:58: warning: Operator Usage Whitespace Violation: Operators should be surrounded by a single whitespace when they are being used. (operator_usage_whitespace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

```swift
// This triggers a violation:
let foo = GenericType<(UIViewController) -> Void>()
// I.e. a generic type that’s typed with a closure when initialized.
// It’s autocorrected to:
let foo = GenericType<(UIViewController) -> Void > ()
```

**Comment by rastersize** [comment]
The issue can be worked around in two ways. Neither of the following examples trigger the bug:

```swift
let foo: GenericType<(UIViewController) -> Void> = .init()
```

or

```swift
typealias SomeClosure = (UIViewController) -> Void
let foo = GenericType<SomeClosure>()
```

**Comment by macdrevx** [comment]
Also happens in situations where a generic parameter itself has a generic parameter:

```
let foo = Foo<Bar<T>, Baz>()
// is autocorrected to
let foo = Foo<Bar<T>, Baz > ()
```

**Comment by marcelofabri** [comment]
Fixed in https://github.com/realm/SwiftLint/pull/2196

**Comment by lazarevzubov** [comment]
There's still an issue with generic parameters, if they're placed on different lines. This:

```swift
let something = Something<GenericParameter1,
                                                GenericParameter2>()
```

Becomes this:

```swift
let something = Something < GenericParameter1,
                                                GenericParameter2 > ()
```

**Comment by SimplyDanny** [comment]
Would you please open a separate issue for that, @lazarevzubov? Herein, it will probably get lost.

**Comment by lazarevzubov** [comment]
> Would you please open a separate issue for that, @lazarevzubov? Herein, it will probably get lost.

Sure, done: https://github.com/realm/SwiftLint/issues/3960
